File: app/kotlin/puyo-puyo-tdd/src/commonTest/kotlin/ConfigTest.kt
package com.example.puyopuyo

import kotlin.test.Test
import kotlin.test.assertEquals

class ConfigTest {
    @Test
    fun ステージの幅が正しく設定されている() {
        // Arrange
        val config = Config()

        // Act & Assert
        assertEquals(6, config.stageWidth)
    }

    @Test
    fun ステージの高さが正しく設定されている() {
        // Arrange
        val config = Config()

        // Act & Assert
        assertEquals(13, config.stageHeight)
    }

    @Test
    fun ぷよのサイズが正しく設定されている() {
        // Arrange
        val config = Config()

        // Act & Assert
        assertEquals(40, config.puyoSize)
    }
}

File: app/typescript-5/puyo-puyo-game/src/models/__tests__/Config.test.ts
import { Config } from '../Config'

describe('Config', () => {
  let config: Config

  beforeEach(() => {
    config = new Config()
  })

  describe('ゲームの基本設定', () => {
    it('ステージの列数が6であること', () => {
      expect(config.stageColumns).toBe(6)
    })

    it('ステージの行数が13であること', () => {
      expect(config.stageRows).toBe(13)
    })

    it('ぷよのサイズが40であること', () => {
      expect(config.puyoSize).toBe(40)
    })

    it('ステージの幅が列数×ぷよサイズであること', () => {
      expect(config.stageWidth).toBe(config.stageColumns * config.puyoSize)
    })

    it('ステージの高さが行数×ぷよサイズであること', () => {
      expect(config.stageHeight).toBe(config.stageRows * config.puyoSize)
    })
  })
})
